File: internal/queue_processor/task.go
package queue_processor

import (
	"github.com/TheAsda/skalka/pkg/config"
	"io"
)

type Task struct {
	name    string
	command string
	stdout  io.Writer
	stderr  io.Writer
	env     config.Env
	dir     string
	runner  Runner
	// TODO: add variables store
}

func NewTask(name string, command string, stdout io.Writer, stderr io.Writer, env config.Env, dir string, runner Runner) *Task {
	return &Task{name: name, command: command, stdout: stdout, stderr: stderr, env: env, dir: dir, runner: runner}
}

func (t *Task) Execute() error {
	err := t.runner.Run(t.command, t.env, t.dir)
	return err
}

File: Skalka/TaskRunner/Task.cs
using System.Collections.Generic;
using System.IO;

namespace Skalka.TaskRunner
{
    public class Task
    {
        public string Name;
        public string Command;
        public Dictionary<string, string> Env;
        public string Dir;

        public Task(string name, string command, Dictionary<string, string> env,
            string dir = null)
        {
            Name = name;
            Command = command;
            Env = env;
            Dir = dir ?? Directory.GetCurrentDirectory();
        }
    }
}